apps/api_web/lib/api_web/params.ex

```diff
@@ -208,6 +208,26 @@ defmodule ApiWeb.Params do
   def canonical("false"), do: false
   def canonical(_), do: nil
 
+  @doc """
+  Parse revenue filter to valid params
+  """
+  def revenue(value) when is_binary(value) do
+    value
+    |> split_on_comma()
+    |> MapSet.new()
+    |> MapSet.intersection(MapSet.new(["REVENUE", "NON_REVENUE"]))
+    |> MapSet.to_list()
+    |> Enum.sort()
+    |> revenue()
+  end
+
+  for value <- ~w(NON_REVENUE REVENUE)a do
+    def revenue([unquote(Atom.to_string(value))]), do: {:ok, unquote(value)}
+  end
+
+  def revenue(["NON_REVENUE", "REVENUE"]), do: {:ok, :ALL}
+  def revenue(_value), do: :error
```

Comment:
```suggestion
```suggestion
  def revenue(values) do
    values
    |> split_on_comma()
    |> Enum.reduce_while({:ok, []}, fn
      "REVENUE", {:ok, acc} -> {:cont, {:ok, [:REVENUE | acc]}}
      "NON_REVENUE", {:ok, acc} -> {:cont, {:ok, [:NON_REVENUE | acc]}}
      _ -> {:halt, :error}
    end)
  end
```